File: backend-spring/src/main/java/com/sourcelens/module/ci/entity/CiDiagnostic.java
package com.sourcelens.module.ci.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("ci_diagnostics")
public class CiDiagnostic {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long projectId;

    private Long scanTaskId;

    private Long repositoryId;

    private String provider;

    private String workflowName;

    private String workflowRunId;

    private Integer runNumber;

    private String branch;

    private String commitSha;

    private String commitMessage;

    /** PENDING / ANALYZING / COMPLETED / FAILED */
    private String status;

    /** success / failure / cancelled / timed_out */
    private String conclusion;

    private String failureSummary;

    /** COMPILE / TEST / DEPENDENCY / LINT / DOCKER / ENV / UNKNOWN */
    private String errorCategory;

    private String rootCause;

    private String relatedFiles;

    private String fixSuggestions;

    private String rawLogSnippet;

    private String diagnosticJson;

    private String errorMessage;

    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Boolean deleted;
}

File: web-console/src/api/ciDiagnostic.ts
import client from './client'
import type { Result } from './client'

export interface CiDiagnostic {
  id: number
  projectId: number
  scanTaskId: number | null
  repositoryId: number | null
  provider: string
  workflowName: string | null
  workflowRunId: string | null
  runNumber: number | null
  branch: string | null
  commitSha: string | null
  commitMessage: string | null
  status: string
  conclusion: string | null
  failureSummary: string | null
  errorCategory: string | null
  rootCause: string | null
  relatedFiles: string | null
  fixSuggestions: string | null
  rawLogSnippet: string | null
  diagnosticJson: string | null
  errorMessage: string | null
  createdBy: number
  createdAt: string
  updatedAt: string
}

export interface PageResult<T> {
  items: T[]
  page: number
  pageSize: number
  total: number
}

export const ciApi = {
  create: (data: {
    projectId: number
    scanTaskId?: number
    repositoryId?: number
    provider?: string
    workflowName?: string
    workflowRunId?: string
    runNumber?: number
    branch?: string
    commitSha?: string
    commitMessage?: string
    conclusion: string
    rawLogSnippet?: string
  }) => client.post<Result<CiDiagnostic>>('/ci-diagnostics', data),

  listByProject: (projectId: number, page = 1, pageSize = 20, status?: string) => {
    const params = new URLSearchParams({ page: String(page), pageSize: String(pageSize) })
    if (status) params.set('status', status)
    return client.get<Result<PageResult<CiDiagnostic>>>(`/projects/${projectId}/ci-diagnostics?${params}`)
  },

  detail: (id: number) =>
    client.get<Result<CiDiagnostic>>(`/ci-diagnostics/${id}`),

  reanalyze: (id: number) =>
    client.post<Result<CiDiagnostic>>(`/ci-diagnostics/${id}/reanalyze`),
}